### PR TITLE
Fix for #8645: Refactor of empty entry handling

### LINKED ISSUE #8645: An empty entry can still be added without warning message (and lead to an error when compiling)
### JabRef version

Latest development branch build (please note build date below)

### Operating system

GNU / Linux

### Details on version and operating system

JabRef 5.6--2022-04-04--dbf921e Linux 4.19.0-20-amd64 amd64  Java 17.0.2  JavaFX 18+12

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Open a library
2. Add an entry (by clicking on the + button, for example)
3. Save the library
4. Close the library

No warning message is displayed (unlike stated in issue #8096), especially not the message of the PR #8218.
And so, the issue #8096 is still present: biber generates an error (`[8337] Utils.pm:411> ERROR - BibTeX subsystem: /tmp/biber_tmp_Qtbu/a9e03cb3043b9e2c02591da84105bb68_26801.utf8, line 36978, syntax error: found ",", expected one of: number, name (entry type, key, field, or macro name), end of entry ("}" or ")") or quoted string ({...} or "...")`)

My analysis:
1) the current definition of an empty entry is probably too strict.
For me, by default, an empty entry is
````
@Article{,
  creationdate = {2022-04-05T10:41:54},
  owner        = {mlep},
}
````
   So, not really an empty entry.
   But, since an empty _entry key_ is enough to lead to a biber error, I suggest _an empty key should be enough to raise the warning message_. 

2) Because you can compile your file after saving (i.e. while keeping your library opened), the biber error is generated before the warning message has a chance to be displayed. Hence, the warning message should be displayed on saving (i.e. not on closing).

3) The feature "Generate keys before saving (for entries without a key)" in `Options -> Preferences -> Citation key generator` is of no help because no key are generated for such an "almost empty entry".

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

### PR DESCRIPTION
**Pull request by xiaoyanghuo**
Fixes #8645

This is group14 from the course "Open Source Development".

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by xiaoyanghuo** [comment]
Although what we submitted was about that thing, it had flaws. Because we don't understand this task very well, we didn't find the relevant dialog in debugging。
image

**Comment by xiaoyanghuo** [comment]
Thank you for your attention!

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +39,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 
 ### Fixed
 
+- We fixed an issue where A particular empty entry can be added without warning message [#8645](https://github.com/JabRef/jabref/issues/8645)
```
Comment (koppor) [inline CHANGELOG.md:42]:
```suggestion
- We fixed an issue where a particular empty entry can be added without warning message. [#8645](https://github.com/JabRef/jabref/issues/8645)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 
 ### Changed
 
+- Now all the entries with no empty citationkey will be detected by "Check integrity",and a test of it has been added.
```
Comment (koppor) [inline CHANGELOG.md:22]:
```suggestion
- Now all the entries with no empty citationkey will be detected by "Check integrity". 
```

"test" is not user-visible. Thus, removed.

**Review by koppor — CHANGES_REQUESTED** [review]
Some small comments.

Otherwise, looks good.

The issue https://github.com/JabRef/jabref/issues/8645 is NOT fixed. The PR at hand just changes the defaults of JabRef, but a user might not want this behavior.

JabRef really (!) has to remove the empty entry before save.

**Comment by koppor** [comment]
A follow-up PR should REALLY add a test case for the BibTeX shown in https://github.com/JabRef/jabref/issues/8645

**Comment by calixtus** [comment]
I personally am against the change of the default value, since a user might just want to test JabRef and would not expect JabRef to change all of his entries by default when saving again. Looking for the right pref option is just annoying for first time users.

Also this PR does not fix the issue.

**Comment by calixtus** [comment]
Test could be reused though.

File: src/main/java/org/jabref/logic/integrity/CitationKeyChecker.java [hunk 3]
```diff
@@ -11,7 +11,8 @@
 import org.jabref.model.strings.StringUtil;
 
 /**
- * Currently only checks the key if there is an author, year, and title present.
+ * In the past only checks the key if there is an author, year, and title present.
+ * Currently checks the key all the time.
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/integrity/CitationKeyChecker.java:15]: Past is not of interest for the current implementation. Past can be viewed in git history.

**Comment by ThiloteE** [comment]
@xiaoyanghuo thank you and your group for this pr, I hope you have learned something from it.

Closing this pr, because as far as I can see, it is in an incomplete state, there is inactivity and I would like to reduce the number of open pull-requests. Feel free to reopen, if you ever feel like wanting to continue working on it.

**Pull request closed without merging**